### ISSUE TITLE
`JsonNormalizer` creates array JSON when object is passed

### ISSUE THREAD

**Issue by boesing** [CuyZ/Valinor#542]
Hey there,

I wonder if it is possible to have valinor keeping the actual type of the object passed to `normalize`.

When I use the following code, I'd expect `{}` from being generated but instead, it returns `[]`:

```php
$normalizer = (new MapperBuilder())->normalizer(Format::json());
var_dump($normalizer->normalize(new stdClass));
// string(2) "[]"
```

I would not expect a type transition to an array just because I pass an object without properties.

**Comment by simPod** [comment]
have you tried withOptions https://www.php.net/manual/en/json.constants.php#constant.json-force-object?

**Comment by simPod** [comment]
Hm, not supported now. Maybe https://github.com/CuyZ/Valinor/issues/542

**Comment by romm** [comment]
Thanks for the report @boesing, this should be fixed with #549.
